urbit: 0.4.5 -> 0.6.0

jglukasik opened the pull request:
###### Motivation for this change

Version was out of date, and install procedure has changed (following https://urbit.org/docs/using/install/ )

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

jglukasik committed:
urbit: 0.4.5 -> 0.6.0

Following https://urbit.org/docs/using/install/

matthewbauer commented:
@grahamcofborg build urbit

matthewbauer reviewed (dismissed):
I think it’s best to avoid bootstrap scripts. Nix should already bootstrap everything for you. We have meson and ninja built so that you shouldn’t have to specify a buildPhase at all.

GrahamcOfBorg commented:
No attempt on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43799&attempt_id=afe5bcfb-2af3-482c-a2ba-bcc3200a4a52)

The following builds were skipped because they don't evaluate on x86_64-darwin: urbit

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

jglukasik committed:
urbit: Don't use bootstrap scripts, no buildPhase

Incorporating feedback from https://github.com/nixos/nixpkgs/pull/43799

Also cleaned up some cruft from previous versions that didn't build with Meson/Ninja.

jglukasik commented:
Thank you for your feedback @matthewbauer . This is my first nixpkgs contribution, I admit to some cargo-culting :slightly_smiling_face: 

I was able to clean this up a bit and avoid using the bootstrap scripts and buildPhase.

globin committed:
urbit: clean-up
